src/TSTransformer/nodes/expressions/transformThisExpression.ts

```diff
@@ -1,14 +1,35 @@
 import luau from "@roblox-ts/luau-ast";
+import assert from "assert";
 import { errors } from "Shared/diagnostics";
 import { SYMBOL_NAMES, TransformState } from "TSTransformer";
 import { DiagnosticService } from "TSTransformer/classes/DiagnosticService";
 import ts from "typescript";
-
+function isInStaticBlockDeclaration(node: ts.Node) {
+	while (!ts.isClassStaticBlockDeclaration(node.parent)) {
+		if (ts.isClassDeclaration(node.parent) || ts.isClassExpression(node.parent)) {
```

Comment:
I think this would also need to check for other things which have an implicit `this` like function declarations and namespaces.

Reply:
alright, i changed it to `getThisExpressionParent` which returns a ` ts.ClassStaticBlockDeclaration | ts.PropertyDeclaration | ts.MethodDeclaration | undefined` 

I decided to also fix #2137 since we're already dealing with static contexts. I don't really know what to do if `node.parent` is a `MethodDeclaration` (or a non-static `PropertyDeclaration`), so it just returns `luau.globals.self`